src/Tribe/Views/V2/View.php

```diff
@@ -2145,4 +2154,92 @@ protected function get_container_data() {
 
 		return $data;
 	}
+
+	protected function should_show_recent_past_events_view() {
+		$show = $this->context->get( 'show_recent_past', true );
+
+		/**
+		 * Filters Whether the Recent Past Events Should Show for all Views.
+		 *
+		 * @since TBD
+		 *
+		 * @param aboolean $show      If we should display Recent Past Events.
+		 * @param string   $view_slug The current view slug.
+		 * @param View     $instance  The current View object.
+		 */
+		$show = apply_filters( 'tribe_events_views_v2_show_recent_past_events_view', $show, $this->get_slug(), $this );
+
+		/**
+		 *  Filters Whether the Recent Past Events Should Show for a specific View.
+		 *
+		 * @since TBD
+		 *
+		 * @param boolean $show     If we should display Recent Past Events.
+		 * @param View    $instance The current View object.
+		 */
+		$show = apply_filters( "tribe_events_views_v2_{$this->get_slug()}_show_recent_past_events_view", $show, $this );
+
+		return $show;
+	}
+
+
+	/**
+	 * @param array $events
+	 * @param array $template_vars
+	 */
+	protected function setup_additional_views( array $events = [], array $template_vars = [] ) {
+		// todo: flatten the events if required (month, week)
+		// todo add threshold and filter
+		if ( 0 === count( $events ) && ! empty( $template_vars['show_recent_past'] ) ) {
+			$template_vars['show_recent_past'] = true;
+			$recent_past_view                  = new Recent_Past_View();
+			$recent_past_view->set_context( $this->context );
+			$instance = $recent_past_view->make( Recent_Past_View::Class );
+			$recent_past_view->template = new Template( $instance );
+			$template_vars['recent_past_view_data'] = $recent_past_view->get_template_vars();
+
+			$whitelist                              = [
```

Comment:
:warning: Equals sign not aligned correctly; expected 1 space but found 30 spaces

<sub>hash: a771dd69b730ccb417247d281329bb12</sub>